### ISSUE TITLE
Removing trailing comma also removes closing parens when next line starts directly with a comment

### ISSUE THREAD

**Issue by gaelrenoux** [scalameta/scalafmt#1228]
OK, I know the title is long and a bit weird, but so is this bug.

- **Version**: 1.6.0-RC3
- **Integration**: SBT or CLI
- **Configuration**: none (all default, .scalafmt.conf is empty)

## Steps

Given code like this:
```scala
class Stuff {
  def id(str: String): String = str
  val fields = Seq(
    id("name"),
// this is a comment
  )
}
```
Notice that the comment starts on the very first character of the line (no space before it).

I run scalafmt through CLI or SBT:
```bash
scalafmt
sbt scalafmt
```

## Problem

Scalafmt formats code like this:
```scala
class Stuff {
  def id(str: String): String = str
  val fields = Seq(
    id("name"
// this is a comment
  )
}
```
A parens has been lost !

## Expectation

I would like the formatted output to look like this:
```scala
class Stuff {
  def id(str: String): String = str
  val fields = Seq(
    id("name")
// this is a comment
  )
}
```

The trailing comma has been removed, but the parens is still there. I don't really care what happens the comment.

## Workaround

In the source file, just having a space before the comment helps.

Or in the configuration:
```
trailingCommas = preserve
```

**Comment by gabro** [comment]
Hu, that's a nasty one, thanks for reporting @gaelrenoux.
I'll try to look into it, @olafurpg.

**Comment by olafurpg** [comment]
Thanks for reporting! Awesome @gabro, heads up scalafmt does special
handling for comments without indentation so I suspect this is the culprit.

**Comment by gabro** [comment]
Pending fix in #1230
